op-chain-ops/genesis/db_migration.go

```diff
@@ -26,28 +92,55 @@ func MigrateDB(db vm.StateDB, config *DeployConfig, l1Block *types.Block, l2Addr
 		return err
 	}
 
-	// Convert all of the messages into legacy withdrawals
-	messages := make([]*crossdomain.LegacyWithdrawal, 0)
-	for _, msg := range migrationData.OvmMessages {
-		wd, err := msg.ToLegacyWithdrawal()
-		if err != nil {
-			return err
-		}
-		messages = append(messages, wd)
+	err = crossdomain.MigrateWithdrawals(withdrawals, db, &l2Addrs.L1CrossDomainMessengerProxy, &l2Addrs.L1StandardBridgeProxy)
+	if err != nil {
+		return err
 	}
-	for _, msg := range migrationData.EvmMessages {
-		wd, err := msg.ToLegacyWithdrawal()
-		if err != nil {
-			return err
-		}
-		messages = append(messages, wd)
+
+	addrs := migrationData.Addresses()
+	if err := ether.MigrateLegacyETH(ldb, addrs, migrationData.OvmAllowances, 0); err != nil {
+		return err
 	}
 
-	if err := crossdomain.MigrateWithdrawals(messages, db, &l2Addrs.L1CrossDomainMessengerProxy, &l2Addrs.L1StandardBridgeProxy); err != nil {
+	if !commit {
+		return nil
+	}
+
+	root, err := db.Commit(true)
+	if err != nil {
 		return err
 	}
 
-	// TODO: use migration data to double check things
+	// Create the bedrock transition block
```

Comment:
The config here could cause problems, I feel like some validation must be done, ie the timestamp could go backwards

Reply:
Do we need this block to have the state transition? I was kinda thinking there wouldn't be one, but this makes sense

Reply:
The root is used to build the genesis output commitment right?

Reply:
hmm, probably. I haven't thought much about it